Remote questions setup fails: preferences.js missing from dist

## Bug

When configuring Telegram remote questions during `gsd config`, the setup fails with:

```
▲  Remote questions setup failed: Cannot find module '/opt/homebrew/lib/node_modules/gsd-pi/dist/resources/extensions/gsd/preferences.js' imported from /opt/homebrew/lib/node_modules/gsd-pi/dist/remote-questions-config.js
```

## Cause

`dist/remote-questions-config.js` (line 12) imports:
```js
import { getGlobalGSDPreferencesPath } from "./resources/extensions/gsd/preferences.js";
```

But only `preferences.ts` exists in the dist — there is no compiled `preferences.js`. The extension `.ts` files are loaded via jiti at runtime, but `remote-questions-config.js` is pre-compiled JS that does a standard ESM import expecting `.js`.

This is likely a partial fix for #592 — the config helper was extracted into compiled JS, but the dependency on `preferences.ts` wasn't also compiled or re-exported from a compiled module.

## Impact

- Telegram (and likely Slack/Discord) remote questions setup fails during onboarding
- The error is non-blocking — GSD continues to work, and the Telegram bot token in `auth.json` still allows sending messages
- But `saveRemoteQuestionsConfig()` never writes the `remote_questions` block to `~/.gsd/preferences.md`

## Workaround

Create a minimal shim at `dist/resources/extensions/gsd/preferences.js`:

```js
import { homedir } from "node:os";
import { join } from "node:path";

const GLOBAL_PREFERENCES_PATH = join(homedir(), ".gsd", "preferences.md");

export function getGlobalGSDPreferencesPath() {
  return GLOBAL_PREFERENCES_PATH;
}
```

## Version

`gsd-pi@2.28.0` (latest as of 2025-03-18)

--- a/src/remote-questions-config.ts
+++ b/src/remote-questions-config.ts
@@ -9,11 +9,17 @@
  */
 
 import { existsSync, readFileSync, writeFileSync, mkdirSync } from "node:fs";
-import { dirname } from "node:path";
-import { getGlobalGSDPreferencesPath } from "./resources/extensions/gsd/preferences.js";
+import { dirname, join } from "node:path";
+import { homedir } from "node:os";
+
+// Inlined from preferences.ts to avoid crossing the compiled/uncompiled
+// boundary — this file is compiled by tsc, but preferences.ts is loaded
+// via jiti at runtime. Importing it as .js fails because no .js exists
+// in dist/. See #592, #1110.
+const GLOBAL_PREFERENCES_PATH = join(homedir(), ".gsd", "preferences.md");
 
 export function saveRemoteQuestionsConfig(channel: "slack" | "discord" | "telegram", channelId: string): void {
-  const prefsPath = getGlobalGSDPreferencesPath();
+  const prefsPath = GLOBAL_PREFERENCES_PATH;
   const block = [
     "remote_questions:",
     `  channel: ${channel}`,

